packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,291 @@
+    if (!node) {
+      return;
+    }
+
+    table.current = node;
+    const resizeObserver = new ResizeObserver(() => {
+      // Update table height state variable
+      setTableHeight(table.current.offsetHeight - 1);
+    });
+
+    // Register resize observer to detect size changes
+    resizeObserver.observe(node);
+  }, []);
+
+  // -------------------------------------
+  // useEffect Hooks
+
+  // useEffect for row displayed columns
+  useEffect(() => {
+    setRenderedColumns(displayedColumns.map((column) => initializeColumn(column)));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pinnedColumns, overflowColumns]);
+
+  // useEffect to calculate pinned column offsets
+  useEffect(() => {
+    const offsetArray = [];
+    if (pinnedColumns.length > 1) {
+      let cumulativeOffset = 0;
+
+      offsetArray.push(cumulativeOffset);
+      const lastPinnedColumnIndex = pinnedColumns.length - 1;
+
+      renderedColumns.slice(0, lastPinnedColumnIndex).forEach((pinnedColumn) => {
+        cumulativeOffset += pinnedColumn.width;
+        offsetArray.push(cumulativeOffset);
+      });
+    }
+    setPinnedColumnOffsets(offsetArray);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [renderedColumns]);
+
+  // -------------------------------------
+
+  // -------------------------------------
+  // event handlers
+
+  const handleColumnSelect = useCallback((columnId) => {
+    if (onColumnSelect) {
+      onColumnSelect(columnId);
+    }
+  }, [onColumnSelect]);
+
+  const onResizeMouseDown = useCallback((event, index, resizeColumnWidth) => {
+    // Store current table and column values for resize calculations
+    tableWidth.current = table.current.offsetWidth;
+    activeColumnPageX.current = event.pageX;
+    activeColumnWidth.current = resizeColumnWidth;
+
+    // Set the active index to the selected column
+    setActiveIndex(index);
+  }, []);
+
+  const onMouseMove = (event) => {
+    if (activeIndex == null) {
+      return;
+    }
+
+    // Ensure the new column width falls within the range of the minimum and maximum values
+    const diffX = event.pageX - activeColumnPageX.current;
+    const { minimumWidth, maximumWidth } = renderedColumns[activeIndex];
+    const newColumnWidth = Math.min(Math.max(activeColumnWidth.current + diffX, minimumWidth), maximumWidth);
+
+    // Update the width for the column in the state variable
+    const newColumns = [...renderedColumns];
+    newColumns[activeIndex].width = newColumnWidth;
+    setRenderedColumns(newColumns);
+
+    // Update the column and table width
+    table.current.style.width = `${tableWidth + (newColumnWidth - activeColumnWidth.current)}px`;
+  };
+
+  const onMouseUp = () => {
+    if (onColumnResize) {
+      onColumnResize(renderedColumns[activeIndex].id, renderedColumns[activeIndex].width);
+    }
+
+    // Remove active index
+    setActiveIndex(null);
+  };
+
+  // -------------------------------------
+
+  return (
+    <div>
```

Comment:
We should probably have a class associated to this container div.  It will need styles.  For example it should have scroll capabilities.  You should be able to reference the container that was around the grid control.  We would want to encapsulate that capability within our control.